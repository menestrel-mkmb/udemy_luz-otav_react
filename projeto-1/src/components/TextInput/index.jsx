export default function TextInput({ searchValue, handleChange }) {
    return (
        <input
            type="search"
            placeholder="Search for post"
            onChange={handleChange}
            value={searchValue}
        />
    );
}