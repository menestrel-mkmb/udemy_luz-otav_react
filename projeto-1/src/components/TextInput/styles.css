.text__inp {
    width: 100%;
    max-width: 800px;
    min-width: 300px;
    
    padding: .5rem 2rem;
}