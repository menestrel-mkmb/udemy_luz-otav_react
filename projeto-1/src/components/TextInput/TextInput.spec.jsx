import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import TextInput from '.';

describe('<TextInput />', () => {
   it('should have a value of searchValue', () => {
    const fn = jest.fn();
    const phText = 'Search for post';
    render(<TextInput searchValue={'testando'} handleChange={fn} />);

    expect(screen.getByPlaceholderText(phText))
        .toBeInTheDocument();
    expect(screen.getByPlaceholderText(phText).value)
        .toBe('testando');
   });

   it('should call handleChange for each key pressed', () => {
    const fn = jest.fn();
    const value = 'testando';
    render(<TextInput handleChange={fn} />);
    
    const input = screen.getByPlaceholderText('Search for post');
    
    expect(input.value).toBe('');
    
    userEvent.type(input, value);

    expect(fn).toHaveBeenCalledTimes(value.length);
    expect(input).toBeInTheDocument();
    expect(input.value).toBe(value);

    userEvent.clear(input);

    expect(input.value).toBe('');
   });
});