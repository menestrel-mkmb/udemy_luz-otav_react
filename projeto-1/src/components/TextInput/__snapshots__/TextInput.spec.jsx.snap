// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TextInput /> should match snapshot 1`] = `
<input
  class="text__inp"
  placeholder="Search for post"
  type="search"
  value=""
/>
`;
